osu.Game/Graphics/Containers/LogoTrackingContainer.cs

```diff
@@ -0,0 +1,160 @@
+        /// </summary>
+        public bool Tracking
+        {
+            get => tracking;
+            set
+            {
+                if (Logo != null)
+                {
+                    if (value && !tracking && Logo.IsTracking)
+                        throw new InvalidOperationException($"Cannot track an instance of {typeof(OsuLogo)} to multiple {typeof(LogoTrackingContainer)}s");
+
+                    Logo.IsTracking = value;
+                }
+
+                tracking = value;
+            }
+        }
+
+        protected OsuLogo Logo;
+
+        private float facadeScale;
+        private Easing easing;
+        private Vector2? startPosition;
+        private double? startTime;
+        private double duration;
+        private bool tracking;
+
+        public LogoTrackingContainer()
+        {
+            LogoFacade = new ExposedFacade();
+        }
+
+        /// <summary>
+        /// Assign the logo that should track the Facade's position, as well as how it should transform to its initial position.
+        /// </summary>
+        /// <param name="logo">The instance of the logo to be used for tracking.</param>
+        /// <param name="facadeScale">The scale of the facade. Does not actually affect the logo itself.</param>
+        /// <param name="duration">The duration of the initial transform. Default is instant.</param>
+        /// <param name="easing">The easing type of the initial transform.</param>
+        public void SetLogo(OsuLogo logo, float facadeScale = 1.0f, double duration = 0, Easing easing = Easing.None)
+        {
+            if (Logo != logo && Logo != null)
+            {
+                // If we're replacing the logo to be tracked, the old one no longer has a tracking container
+                Logo.IsTracking = false;
+            }
+
+            Logo = logo ?? throw new ArgumentNullException(nameof(logo));
+
+            if (Tracking)
+            {
+                Logo.IsTracking = true;
+            }
+
+            this.facadeScale = facadeScale;
+            this.duration = duration;
+            this.easing = easing;
+
+            startTime = null;
+            startPosition = null;
+        }
+
+        /// <summary>
+        /// Gets the position that the logo should move to with respect to the <see cref="LogoFacade"/>.
+        /// Manually performs a conversion of the Facade's position to the Logo's parent's relative space.
+        /// </summary>
+        /// <remarks>Will only be correct if the logo's <see cref="Drawable.RelativePositionAxes"/> are set to Axes.Both</remarks>
+        protected Vector2 LogoTrackingPosition => new Vector2(Logo.Parent.ToLocalSpace(LogoFacade.ScreenSpaceDrawQuad.Centre).X / Logo.Parent.RelativeToAbsoluteFactor.X,
+            Logo.Parent.ToLocalSpace(LogoFacade.ScreenSpaceDrawQuad.Centre).Y / Logo.Parent.RelativeToAbsoluteFactor.Y);
```

Comment:
You're doing two matrix multiplications per query of this. Compute `Logo.Parent.ToLocalSpace(...)` in a local variable before returning.